src/shared/pretty-print.c

```diff
@@ -531,28 +531,24 @@ void clear_progress_bar_unbuffered(const char *prefix) {
 }
 
 void draw_progress_bar(const char *prefix, double percentage) {
-
         /* We are going output a bunch of small strings that shall appear as a single line to STDERR which is
          * unbuffered by default. Let's temporarily turn on full buffering, so that this is passed to the tty
          * as a single buffer, to make things more efficient. */
-        char buffer[LONG_LINE_MAX];
-        setvbuf(stderr, buffer, _IOFBF, sizeof(buffer));
-
+        WITH_BUFFERING(stderr);
         draw_progress_bar_unbuffered(prefix, percentage);
```

Comment:
Not your fault, but I'd like to rename the _unbuffered funcs to _impl, which actually don't care about the buffer